src/Stratis.Features.Unity3dApi/Controllers/Unity3dController.cs

```diff
@@ -511,6 +516,32 @@ public LocalExecutionResult LocalCallSmartContractTransaction([FromBody] LocalCa
             return result as LocalExecutionResult;
         }
 
+        /// <summary>
+        /// Searches a smart contract's receipts for those which match a specific event. The SmartContract.Log() function
+        /// is capable of storing C# structs, and structs are used to store information about different events occurring 
+        /// on the smart contract. For example, a "TransferLog" struct could contain "From" and "To" fields and be used to log
+        /// when a smart contract makes a transfer of funds from one wallet to another. The log entries are held inside the smart contract,
+        /// indexed using the name of the struct, and are linked to individual transaction receipts.
+        /// Therefore, it is possible to return a smart contract's transaction receipts
+        /// which match a specific event (as defined by the struct name).  
+        /// </summary>
+        /// 
+        /// <param name="contractAddress">The contract address from which events were raised.</param>
+        /// <param name="eventName">The name of the event raised.</param>
+        /// <param name="topics">The topics to search. All specified topics must be present.</param>
+        /// <param name="fromBlock">The block number from which to start searching.</param>
+        /// <param name="toBlock">The block number where searching finishes.</param>
+        /// 
+        /// <returns>A list of receipts for transactions relating to a specific smart contract and a specific event in that smart contract.</returns>
+        [Route("api/[controller]/receipt-search")]
+        [HttpGet]
+        public async Task<List<ReceiptResponse>> ReceiptSearchAPI([FromQuery] string contractAddress, [FromQuery] string eventName, [FromQuery] List<string> topics = null, [FromQuery] int fromBlock = 0, [FromQuery] int? toBlock = null)
+        {
+            List<ReceiptResponse> result = this.smartContractsController.ReceiptSearch(contractAddress, eventName, topics, fromBlock, toBlock);
```

Comment:
Hi Ivan,

It seems that the logic in `SmartContractsController.ReceiptSearch` is now shared between 2 controllers. There is already a `smartContractTransactionService` in this controller, so move the logic in `SmartContractsController.ReceiptSearch` there and then just call `smartContractTransactionService.ReceipSearch` from here (also just update `SmartContractsController` to do the same.

😎

Reply:
`It seems that the logic in SmartContractsController.ReceiptSearch is now shared between 2 controllers.`
Logic was just in SC controller and unity controller called it (same was done with other methods as well)

But anyway, changes applied